sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/sarif/SarifParserCallbackImplTest.java

```diff
@@ -82,25 +83,36 @@ public void should_add_project_issues() {
   }
 
   @Test
-  public void should_add_file_issues() {
+  public void should_add_file_issues_with_secondary_location() {
     String absoluteFilePath = temp.getRoot().toPath().resolve("file1").toString();
-    callback.onFileIssue("rule1", "warning", absoluteFilePath, "msg");
+    callback.onFileIssue("rule1", "warning", absoluteFilePath, Collections.singletonList(createLocation("file1", 4, 5)), "msg");
     assertThat(ctx.allIssues()).hasSize(1);
     assertThat(ctx.allIssues().iterator().next().primaryLocation().inputComponent().key()).isEqualTo("module1:file1");
     assertThat(ctx.allIssues().iterator().next().ruleKey().rule()).isEqualTo("rule1");
     assertThat(logTester.logs(LoggerLevel.DEBUG)).containsOnly("Adding file level issue rule1: file1");
+
+    List<Flow> flows = ctx.allIssues().iterator().next().flows();
+    assertThat(flows).hasSize(1);
+    List<IssueLocation> locations = flows.get(0).locations();
+    assertThat(locations).hasSize(1);
+    assertThat(locations.get(0).inputComponent().key()).isEqualTo("module1:file1");
+    TextRange textRange = locations.get(0).textRange();
+    assertThat(textRange.start().lineOffset()).isEqualTo(5);
+    assertThat(textRange.start().line()).isEqualTo(4);
+    assertThat(textRange.end().lineOffset()).isEqualTo(6);
+    assertThat(textRange.end().line()).isEqualTo(4);
   }
 
   @Test
   public void should_create_external_file_issue_for_unknown_rule_key() {
     String absoluteFilePath = temp.getRoot().toPath().resolve("file1").toString();
-    callback.onFileIssue("rule45", "warning", absoluteFilePath, "msg");
+    callback.onFileIssue("rule45", "warning", absoluteFilePath, emptyList(), "msg");
```

Comment:
could you please add one more test with secondary locations for external issues?